backend/typescript/rest/givecloud.ts

```diff
@@ -0,0 +1,58 @@
+import { Router } from "express";
+import AuthService from "../services/implementations/authService";
+import UserService from "../services/implementations/userService";
+import { sendErrorResponse } from "../utilities/errorResponse";
+import { AuthDTO, Role, UserDTO } from "../types";
+import { isAuthorizedByRole } from "../middlewares/auth";
+
+const givecloudRouter: Router = Router();
+
+const authDtoToToUserDto = (authDTO: AuthDTO): UserDTO => {
+  const userDTO: UserDTO = (({
+    id,
+    firstName,
+    lastName,
+    email,
+    roleType,
+    subscriptionExpiresOn,
+  }) => ({ id, firstName, lastName, email, roleType, subscriptionExpiresOn }))(
+    authDTO,
+  );
+  return userDTO;
+};
+
+const userService = new UserService();
+const authService = new AuthService(userService);
+givecloudRouter.use(isAuthorizedByRole(new Set(["Admin"])));
+
+givecloudRouter.post("/user.subscription_paid", async (req, res) => {
+  try {
+    const { membership, supporter } = req.body;
```

Comment:
From the example response, isn't "membership" nested within the "supporter" object? This might be why I'm getting the "cannot read property" error